src/Module/Contact.php

```diff
@@ -759,14 +758,28 @@ public static function content(array $parameters = [], $update = 0)
 			case 'mutuals': $sql_extra .= " AND `rel` = 3"; break;
 		}
 
+		$sql_join_group = '';
+		$sql_extra_group = '';
+		if ($group) {
+			$sql_join_group = "LEFT JOIN `group_member` ON `group_member`.`contact-id` = `contact`.`id`";
+			$sql_extra_group = sprintf(" AND `group_member`.`gid` = %d ", DBA::escape($group));
+		}
+
 		$sql_extra .=  " AND NOT `deleted` ";
 
 		$sql_extra2 = ((($sort_type > 0) && ($sort_type <= Model\Contact::FRIEND)) ? sprintf(" AND `rel` = %d ", intval($sort_type)) : '');
 
 		$sql_extra3 = Widget::unavailableNetworks();
 
-		$r = q("SELECT COUNT(*) AS `total` FROM `contact`
-			WHERE `uid` = %d AND `self` = 0 $sql_extra $sql_extra2 $sql_extra3",
+		$r = q("SELECT COUNT(*) AS `total`
```

Comment:
Maybe you can use `DBA::p()` so we lower the dependencies to `include\dba.php`

Reply:
The whole damn thing needs to be rewritten and split between the contact list page and the single contact display page.